pacta/build_with_tag.sh

```diff
@@ -1,51 +1,80 @@
 #! /bin/bash
 
-tag="${1:-latest}"
-clones="${2:-PACTA_analysis create_interactive_report StressTestingModelDev pacta-data}"
+# Examples:
+# build_with_tag 0.0.0.999
+
+repos="PACTA_analysis create_interactive_report StressTestingModelDev pacta-data"
+user_results="user_results"
 url="git@github.com:2DegreesInvesting/"
 
-for repo in ${clones}
-do
-    remote="${url}${repo}.git"
-    git clone -b master ${remote} --depth 1
-    echo "--"
-done
+tag="$1"
+if [ -z "$tag" ]
+then
+    echo "Please give a tag."
+    exit 0
+fi
 
-for repo in ${clones}
-do
-    git -C "${repo}" tag -a "${tag}" -m "Release pacta ${tag}" HEAD
-    echo "${repo}"
-    echo "$(git -C ${repo} log --pretty='%h %d <%an> (%cr)' | head -n 1)"
-    echo "--"
-done
+here="$(basename $(pwd))"
+if [ ! "$here" == "pacta" ]
+then
+    echo "Please run from 2diidockerrunner/pacta (not $(pwd))"
+    exit 0
+fi
+
+clone_and_log () {
+    for repo in $repos
+    do
+        remote="${url}${repo}.git"
+        git clone -b master "$remote" --depth 1 || exit 0
+        echo
+
+        if [ -n "$tag" ]
+        then
+            echo "Tagging as $tag"
+            git -C "$repo" tag -a "$tag" -m "Release pacta $tag" HEAD || exit 0
+            echo
+        fi
+
+        echo "$(git -C $repo log --pretty='%h %d <%an> (%cr)' | head -n 1)"
+        echo
+    done
+}
+
+clone_and_log
 
 docker rmi --force $(docker images -q '2dii_pacta' | uniq)
-docker build --tag 2dii_pacta:"${tag}" --tag 2dii_pacta:latest .
+echo
+
+docker build --tag 2dii_pacta:"$tag" --tag 2dii_pacta:latest .
+echo
 
-for repo in ${clones}
+for repo in $clones
```

Comment:
No big deal here -- just for consistency

Reply:
does `clones` exist anymore?